### ISSUE TITLE
twisted.python.usage.Options's __hash__ is overly complicated

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7165]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Alex reported|
|-|-|
|Trac ID|trac#7165|
|Type|enhancement|
|Created|2014-04-16 01:00:35Z|
|Branch|https://github.com/twisted/twisted/tree/hashes-7165|

Really it's just trying to implement the default object __hash__. This has the added benefit of fixing its Python3 compatibility.

<details><summary>Searchable metadata</summary>

```
trac-id__7165 7165
type__enhancement enhancement
reporter__Alex Alex
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_hashes_7165 branches/hashes-7165
branch_author__pahan pahan
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1397610035000000 1397610035000000
changetime__1397614905000000 1397614905000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Alex's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|Alex commented|
|-|-|

This is resolved by https://github.com/twisted/twisted/pull/20

**Comment by twisted-trac** [comment]
|<img alt="pahan's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|pahan commented|
|-|-|

(In [42327]) Branching to hashes-7165.

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|PenguinOfDoom commented|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/hashes-7165

**Comment by twisted-trac** [comment]
|<img alt="pahan's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|pahan set status to `closed`|
|-|-|

(In [42331]) Merge hashes-7165

Author: Alex
Reviewer: PenguinOfDoom
Fixes: [#7165](https://github.com/twisted/twisted/issues/7165)
